File: 1. Print address of a variable.py
"""
Topic: Print address of a variable using a pointer-like reference
Description: Python doesn't expose raw pointers; use id() to show object address.
I/O: prints to stdout
"""
def main():
    x = 42
    print('Value:', x)
    print('Address (id):', hex(id(x)))

if __name__ == '__main__':
    main()

# Time Complexity: O(1) (constant-time demonstration).
# Space Complexity: O(1) additional space.

File: 1. Print address of a variable.cpp
// Topic: Print address of a variable using a pointer
// Description: Show variable value and its memory address via pointer
// I/O: none, prints to stdout
#include <iostream>
using namespace std;

int main()
{
    int x = 42;
    int* p = &x;  // memoization marker not applicable
    cout << "Value: " << x << '\n';
    cout << "Address (via pointer): " << p << '\n';
    cout << "Address (literal &x): " << &x << '\n';
    return 0;
}

// Time Complexity: O(1) (constant-time demonstration).
// Space Complexity: O(1) additional space.
